app/models/book.rb

```diff
@@ -1,2 +1,9 @@
 class Book < ActiveRecord::Base
+  monetize :price_cents
+
+  
+
+  scope :by_published_date, -> { order(published_date: :desc) } 
+  scope :filter, -> (search) { where("title LIKE ?", "%#{search}%") } 
```

Comment:
I could totally be wrong but it looks like you're sorting the books twice. By having `|| "published_date desc"` in your controller, you shouldn't also need to run it here. If you ended up with a butt ton of books, this could take a long butt time. Comment out line 6 and see if all your tests pass still (they should). If they do, get rid of that sucker.